### PR TITLE
Remove pthread header from SVMLightOneClass

### PR DESCRIPTION
**Pull request by abhinavrai44**

### CODE REVIEW

File: src/shogun/classifier/svm/SVMLightOneClass.cpp [hunk 1]
```diff
@@ -33,10 +33,6 @@ extern "C" {
 
 #include <shogun/base/Parallel.h>
 
-#ifdef HAVE_PTHREAD
```
Comment (karlnapf) [inline src/shogun/classifier/svm/SVMLightOneClass.cpp]: this should be a sep. PR

File: src/shogun/machine/DistanceMachine.cpp [hunk 2]
```diff
@@ -51,168 +43,69 @@ void CDistanceMachine::init()
...
 void CDistanceMachine::distances_lhs(float64_t* result,int32_t idx_a1,int32_t idx_a2,int32_t idx_b)
 {
-    // TODO: port to use OpenMP backend instead of pthread
-#ifdef HAVE_PTHREAD
-    int32_t num_threads=parallel->get_num_threads();
-#else
-    int32_t num_threads=1;
-#endif
-    ASSERT(num_threads>0)
+    int32_t num_threads;
```
Comment (karlnapf) [inline src/shogun/machine/DistanceMachine.cpp]: why do we need a variable for the number of threads?

File: src/shogun/machine/DistanceMachine.cpp [hunk 3]
```diff
@@ -51,168 +43,69 @@ void CDistanceMachine::init()
...
         {
-            params[t].d = distance;
-            params[t].r = result;
-            params[t].idx_r_start=t*step;
-            params[t].idx_start = (t*step)+idx_a1;
-            params[t].idx_stop = ((t+1)*step)+idx_a1;
-            params[t].idx_comp=idx_b;
-
-            pthread_create(&threads[t], &attr, CDistanceMachine::run_distance_thread_lhs, (void*)&params[t]);
+            num_threads=omp_get_num_threads();
```
Comment (karlnapf) [inline src/shogun/machine/DistanceMachine.cpp]: @vigsterkr can you comment on the use of `Parallel` here?

File: src/shogun/machine/DistanceMachine.cpp [hunk 4]
```diff
@@ -51,168 +43,69 @@ void CDistanceMachine::init()
...
-            pthread_join(threads[t], NULL);
+        int32_t idx_r_start = thread_num * step;
+        int32_t idx_start = (thread_num * step) + idx_a1;
+        int32_t idx_stop = (thread_num==num_threads) ? (idx_a2 + 1) : ((thread_num + 1) * step) + idx_a1;
 
-        pthread_attr_destroy(&attr);
-        SG_FREE(params);
-        SG_FREE(threads);
+        for(int32_t i=idx_r_start; idx_start < idx_stop; i++,idx_start++)
+            result[i] = distance->distance(idx_start,idx_b);
```
Comment (karlnapf) [inline src/shogun/machine/DistanceMachine.cpp]: I wonder whether the distance matrix computation should be part of `CDistance` and then we just call that method from here?

File: src/shogun/machine/gp/Inference.cpp [hunk 5]
```diff
@@ -39,18 +39,11 @@
 #include <shogun/mathematics/Math.h>
 #include <shogun/lib/Lock.h>
 
-using namespace shogun;
+#ifdef HAVE_OPENMP
```
Comment (karlnapf) [inline src/shogun/machine/gp/Inference.cpp]: can you please do every class in a separate PR?

File: src/shogun/machine/DistanceMachine.cpp (hunk 3 above)
Comment (abhinavrai44) [inline src/shogun/machine/DistanceMachine.cpp]: @karlnapf The variable involved need to be computed only once as they will remain constant over all the threads. In parallel the same computations will be repeated over all threads.

File: src/shogun/classifier/svm/SVMLightOneClass.cpp (hunk 1 above)
Comment (abhinavrai44) [inline src/shogun/classifier/svm/SVMLightOneClass.cpp]: ok.

File: src/shogun/machine/gp/Inference.cpp (hunk 5 above)
Comment (abhinavrai44) [inline src/shogun/machine/gp/Inference.cpp]: Ok :-)

**Comment by vigsterkr** [comment]
dunno what's happening with travis but i think this one is really safe to merge....
